### PR TITLE
[ BB2-831 ] Setup Terraform for BFD-Insights BB2 custom Configurations

### PR DESCRIPTION
**Pull request by oragame**
**JIRA Ticket:**
[BB2-831](https://jira.cms.gov/browse/BB2-831)

**User Story or Bug Summary:**

Create custom BB2 firehose setup for BFD insights. Modular design for simple setup of new firehose data stream resources.

---

### What Does This PR Do?

* This new config for custom firehose delivery streams replaces the contents of the root  dir of `/insights/terraform/projects/bb2` and the older generic config has been moved into `/insights/terraform/projects/bb2/base_config`
* The new config allows for the custom requirements needed for the BB2 firehose delivery stream. Mainly the `data_format_conversion_configuration` and `processing_configuration` on the `aws_kinesis_firehose_delivery_stream` resource
* Other additional resources included for `cloudwatch log destination` and `glue crawler`
* Modular setup allows for adding additional/future shared resources for this type of firehose delivery stream

### What Should Reviewers Watch For?

* Is this likely to address the goals expressed in the user story?
* Are any additional documentation updates needed?
* Are there any unhandled and/or untested edge cases you can think of?
* Is user input properly sanitized & handled?
* Should any additional resources be added for this initial setup? (Lambda etc.)
* Is the use of Terraform v0.13.7 acceptable? If not, will need to re-factor to allow this to work in a lower version. The `for_each` meta-argument is not available for modules in versions lower than v0.13
* `error_output_prefix` is pointed to the generic `events-errors` in the firehose module. Would it be desirable to have a custom error output?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* Ideally, this PR and the [accompanying PR](https://github.com/CMSgov/bluebutton-web-deployment/pull/1457) on the BB2 repo will be merged and applied at the same time

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by njdister — CHANGES_REQUESTED** [review]
As we discussed, we should move this code from the top level into the projects folder, and archive the old unused BB2 code there.

**Comment by oragame** [comment]
> As we discussed, we should move this code from the top level into the projects folder, and archive the old unused BB2 code there.

Thanks for the review - the code has been moved to the original projects/bb2 directory, and I moved the older existing tf into a `base_config` directory for now.

**Review by njdister — DISMISSED** [review]
Thanks for the restructuring, looking good!

File: insights/terraform/projects/bb2/base_config/main.tf [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  description     = "Raw BB2 events"
+  bucket          = module.bucket.id
+  bucket_cmk      = module.bucket.bucket_cmk
+  tags            = local.tags
+  partitions      = local.partitions
+  columns         = local.common_columns
+}
+
+
+## Firehose for putting simulated data into the events table
```
Comment (dtisza1) [inline insights/terraform/projects/bb2/base_config/main.tf:78]: Could remove/change this comment.

**Review by dtisza1 — DISMISSED** [review]
This looks good to me!

@oragame Great work on this!

Testing went well yesterday and worked as expected!

Just needing a PROD section added and one comment could be removed.

File: insights/terraform/projects/bb2/base_config/main.tf (hunk 1 above)
Comment (oragame) [reply insights/terraform/projects/bb2/base_config/main.tf]: I removed this comment

**Review by njdister — APPROVED** [review]

**Pull request merged**